psycopg/psycopg/types/geometry.py

```diff
@@ -0,0 +1,86 @@
+"""
+Adapters for PostGIS geometries
+"""
+
+from typing import Optional, Type
+
+from .. import postgres
+from ..abc import AdaptContext
+from ..adapt import Dumper, Loader
+from ..pq import Format
+from .._typeinfo import TypeInfo
+
+
+try:
+    import shapely.wkb as wkb
+    from shapely.geometry.base import BaseGeometry
+
+except ImportError:
+    raise ImportError(
+        "The module psycopg.types.geometry requires the package 'Shapely'"
+        " to be installed"
+    )
+
+
+class GeometryBinaryLoader(Loader):
+    format = Format.BINARY
+
+    def load(self, data: bytes) -> "BaseGeometry":
+        return wkb.loads(data)
```

Comment:
I've made a few manual tests and this doesn't seem to work.

The input data should be `psycopg.abc.Buffer`, which is `Union[bytes,bytearray,memoryview]`. Most often, the parameters from a query are passed as `memoryview`. It doesn't seem that wkb.loads accepts bytearray and memoryview, so you should convert this to bytes:
```diff
--- a/psycopg/psycopg/types/geometry.py
+++ b/psycopg/psycopg/types/geometry.py
@@ -5,7 +5,7 @@ Adapters for PostGIS geometries
 from typing import Optional, Type
 
 from .. import postgres
-from ..abc import AdaptContext
+from ..abc import AdaptContext, Buffer
 from ..adapt import Dumper, Loader
 from ..pq import Format
 from .._typeinfo import TypeInfo
@@ -25,14 +25,18 @@ except ImportError:
 class GeometryBinaryLoader(Loader):
     format = Format.BINARY
 
-    def load(self, data: bytes) -> "BaseGeometry":
+    def load(self, data: Buffer) -> "BaseGeometry":
+        if not isinstance(data, bytes):
+            data = bytes(data)
         return wkb.loads(data)
```

Reply:
Fixed, thanks